lib/lexer.mll

```diff
@@ -52,7 +58,7 @@ and cram_text section = parse
         newline lexbuf;
         `Section section :: cram_text (Some section) lexbuf }
   | "  " ([^'\n']* as first_line) eol
-      { let header = Syntax.cram_default_header in
+      { let header = Some Block.Header.Shell in
```

Comment:
It feels weird to have hardcoded headers interpreted internally by `mk`. What about having `Block.mk_shell` that would skip checking the header ?
If the header doesn't make sense in this case, it shouldn't be in `Block.t`.

Reply:
I moved the header field to blocks where it can't be guessed, but I'm not convinced we need a `mk_shell` function, as `mk` performs a lot of checks and I think it's a clear interface as it is with a single constructor function.

Reply:
The header is just a syntax marker for the markdown interpreters that doesn't have a semantic value for MDX. It is used, in addition with other things, such as the syntax of the block content (`$` markers for cram blocks, `#` markers for toplevel blocks, etc...) to infer the type of MDX block we're dealing with. I think it's indeed `mk`'s job to interpret it and we should stick to a minimal parsing here.